### PR TITLE
Add Storage::Bucket#location_type

### LINKED ISSUE #3302: Support Storage::Bucket#location_type
Location type will be a new field added for Bucket resource. It will be populated on-the-fly for output path, and is a read-only field so patch/update/create will not have any effect on bucket resource.

### PR DESCRIPTION
**Pull request by quartzmo**
[closes #3302]

### CODE REVIEW

**Comment by frankyn** [comment]
Please wait to merge until pinged. LGTM. Thanks for surfacing this change so quickly @quartzmo!

**Comment by quartzmo** [comment]
@frankyn Currently, [documentation for `storage_class`](https://github.com/googleapis/google-cloud-ruby/blob/cc3fcfad11862a42c4a1b29f905e9ee23f59fae3/google-cloud-storage/lib/google/cloud/storage/project.rb#L296-L304) contains:

> Accepted values include `:multi_regional`, `:regional`, `:nearline`, and `:coldline`

Should such documentation be changed everywhere to:

> Accepted values include `:standard`, `:nearline`, and `:coldline`

**Comment by JesseLovelace** [comment]
@quartzmo I think it should be changed, perhaps with a quick note about the other fields being deprecated and what to use instead.

Note: "MULTI_REGION", "REGION", and" DUAL_REGION" were supposed to be "multi-region", "region", and "dual-region". This will be changed in the API after a rollout on June 26 and should also be reflected here. (Adding to the issue as well)

**Comment by crwilcox** [comment]
@frankyn can we get an update on the status of this?

**Comment by crwilcox** [comment]
My mistake @JesseLovelace has taken over this.

**Comment by quartzmo** [comment]
I updated this PR to expect "multi-region", "region", and "dual-region". Unit tests are passing, but integration tests currently fail with:

```
  4) Failure:
Storage::buckets::storage#test_0001_gets pages of buckets [/Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-storage/acceptance/storage/buckets_test.rb:35]:
Expected: "multi-region"
  Actual: "MULTI_REGION"

  5) Failure:
Google::Cloud::Storage::Bucket::storage#test_0002_knows its attributes [/Users/quartzmo/code/google/codez/google-cloud-ruby/google-cloud-storage/acceptance/storage/bucket_test.rb:81]:
Expected: "multi-region"
  Actual: "MULTI_REGION"
```

**Comment by quartzmo** [comment]
The tests above are still failing with the same errors today.

**Comment by quartzmo** [comment]
> This will be changed in the API after a rollout on June 26

@JesseLovelace The integration tests still fail today (June 27) with:

```
Expected: "multi-region"
  Actual: "MULTI_REGION"
```

**Comment by JesseLovelace** [comment]
@quartzmo Thank you for letting me know, I will check to find out when we can expect this to be fixed

**Comment by JesseLovelace** [comment]
@quartzmo The plan is now for the rollout to be completed tomorrow

File: google-cloud-storage/lib/google/cloud/storage/bucket.rb [hunk 1]
```diff
@@ -285,6 +285,15 @@ def location
           @gapi.location
         end
 
+        ##
+        # The type of the bucket location.
+        #
+        # @return [String]
+        #
```
Comment (JesseLovelace) [inline google-cloud-storage/lib/google/cloud/storage/bucket.rb]: Please add some documentation here about what this field is and the accepted values

Comment (quartzmo) [reply google-cloud-storage/lib/google/cloud/storage/bucket.rb]: Added.

**Comment by quartzmo** [comment]
@JesseLovelace Integration tests are now passing for this feature. (Meaning that the service is now returning "multi-region" rather than "MULTI_REGION".)

**Comment by JesseLovelace** [comment]
Thanks! Please hold off on merging until I confirm prod rollout

**Comment by quartzmo** [comment]
If you need a release before Monday, July 8, please let us know. Otherwise we will wait until after the holiday weekend.

**Comment by JesseLovelace** [comment]
This can now be merged

**Comment by quartzmo** [comment]
@JesseLovelace Is it also OK to release this feature now?

**Comment by JesseLovelace** [comment]
Yes 🌈

**Pull request merged**